py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,797 @@
+            'c_max': round(c_data['temp_max']),
+
+            'f': round(f_data['temp']),
+            'f_min': round(f_data['temp_min']),
+            'f_max': round(f_data['temp_max']),
+
+            'k': round(k_data['temp']),
+            'k_min': round(k_data['temp_min']),
+            'k_max': round(k_data['temp_max']),
+        })
+
+    def _format_sunrise(self, wthr):
+        # Get the time for sunrise (default is the start of time)
+        dt = datetime.datetime.utcfromtimestamp(
+            self._jpath(wthr, OWM_SUNRISE, 0))
+
+        # Format the sunrise
+        replaced = dt.strftime(self.format_sunrise)
+        return self.py3.safe_format(replaced, {
+            'icon': self.icons['sunrise'],
+        })
+
+    def _format_sunset(self, wthr):
+        # Get the time for sunset (default is the start of time)
+        dt = datetime.datetime.utcfromtimestamp(
+            self._jpath(wthr, OWM_SUNSET, 0))
+
+        # Format the sunset
+        replaced = dt.strftime(self.format_sunset)
+        return self.py3.safe_format(replaced, {
+            'icon': self.icons['sunset'],
+        })
+
+    def _format_dict(self, wthr):
+        return {
+            # Standard options
+            'icon': self._get_icon(wthr),
+            'clouds': self._format_clouds(wthr),
+            'rain': self._format_rain(wthr),
+            'snow': self._format_rain(wthr),
+            'wind': self._format_wind(wthr),
+            'humidity': self._format_humidity(wthr),
+            'pressure': self._format_pressure(wthr),
+            'temp': self._format_temp(wthr),
+            'sunrise': self._format_sunrise(wthr),
+            'sunset': self._format_sunset(wthr),
+
+            # Descriptions (defaults to empty)
+            'desc': self._jpath(wthr, OWM_DESC, ''),
+            'desc_long': self._jpath(wthr, OWM_DESC_LONG, '')
+        }
+
+    def _format(self, wthr, fcsts):
+        # Format all sections
+        today = self._format_dict(wthr)
+
+        # Insert forecasts
+        forecasts = []
+        for day in fcsts:
+            future = self._format_dict(day)
+            forecasts.append(self.py3.safe_format(self.format_forecast, future))
+
+        # Give the final format
+        today['forecast'] = self.py3.composite_join(
+            self.forecast_text_separator, forecasts)
+        return self.py3.safe_format(self.format, today)
+
+    def weather(self):
+        # Get weather information
+        coords = self._get_coords()
+        text = ''
+        if (coords is not None):
+            wthr = self._get_weather(coords)
+            fcsts = self._get_forecast(coords)
+
+            text = self._format(wthr, fcsts)
+
+        return {
+            'full_text': text,
+            'cached_until': self.py3.time_in(seconds=self.cache_timeout)
+        }
+
+    def on_click(self, event):
+        # Avoid hitting rate limits on APIs
+        self.py3.prevent_refresh()
+
+
+if (__name__ == '__main__'):
```

Comment:
These parentheses are unneeded and should be removed this seems to be something you are doing quite a lot, it makes the code look like JavaScript :p

We should use them only where it clarifies things, here it just makes the code harder to read.